src/js/components/account/user-registration.js

```diff
@@ -84,11 +86,18 @@ export class UserRegistration {
       const $formElementsContainer = $("<div />");
       $formElementsContainer.append(getLabel(field.label));
 
-      if (field.label === "password") {
+      if (field.label.toLowerCase() === "password") {
         $formElementsContainer.append(
           getInput(field.type, field.label, "", "password-requirements")
         );
         $formElementsContainer.append(getPasswordRequirements());
+      } else if (
+        field.label.toLowerCase() === "confirm password" ||
+        field.label.toLowerCase() === "email address"
+      ) {
+        $formElementsContainer.append(
+          getInput(field.type, field.label, "", null, field.name)
+        );
```

Comment:
let's use name rather than label just to fix this bug, but I think we should be thinking about more declarative forms rather than logic to decide which fields should be shown

Reply:
Indeed, this turned into a beast of yuk. I do want to refactor this. But agreed, let's get the bug fix and then refactor.